httpx/_client.py

```diff
@@ -1521,6 +1532,7 @@ async def send(
             for hook in self._event_hooks["response"]:
                 await hook(response)
 
+            self.pending_requests -= 1
```

Comment:
Really we'd want to decrement pending requests, at the point the response is closed.

(This functionality might be more fiddly than it looks at first.)